includes/class-newspack-popups.php

```diff
@@ -388,5 +392,50 @@ public static function popup_default_fields( $post_id, $post, $update ) {
 		update_post_meta( $post_id, 'trigger_scroll_progress', 30 );
 		update_post_meta( $post_id, 'utm_suppression', '' );
 	}
+
+	/**
+	 * Create the config file for the API, unless it exists.
+	 */
+	public static function create_lightweight_api_config() {
+		if ( get_option( 'newspack_has_tried_to_create_lightweight_api_config' ) ) {
+			return;
+		};
+		add_option( 'newspack_has_tried_to_create_lightweight_api_config', true );
+		$has_db_config_in_env = getenv( 'DB_USER' ) && getenv( 'DB_PASSWORD' ) && getenv( 'DB_NAME' ) && getenv( 'DB_HOST' );
+		if ( ! $has_db_config_in_env || file_exists( self::LIGHTWEIGHT_API_CONFIG_FILE_PATH ) ) {
+			return;
+		}
+		global $wpdb;
+		file_put_contents(
+			self::LIGHTWEIGHT_API_CONFIG_FILE_PATH,
+			'<?php' .
+			// Insert these only if they are defined, but not in the as environment variables.
+			// This way only variables which are already declared in wp-config.php should be inserted in this config file.
+			( ! getenv( 'DB_CHARSET' ) && defined( 'DB_CHARSET' ) ? "\ndefine( 'DB_CHARSET', '" . DB_CHARSET . "' );" : '' ) .
+			( ! getenv( 'WP_CACHE_KEY_SALT' ) && defined( 'WP_CACHE_KEY_SALT' ) ? "\ndefine( 'WP_CACHE_KEY_SALT', '" . WP_CACHE_KEY_SALT . "' );" : '' ) .
+			"\ndefine( 'DB_PREFIX', '" . $wpdb->prefix . "' );" .
+			"\n"
+		);
+		error_log( 'Created the config file: ' . self::LIGHTWEIGHT_API_CONFIG_FILE_PATH );
+	}
+
+	/**
+	 * Add an admin notice if config is missing.
+	 */
+	public static function api_config_missing_notice() {
+		if ( file_exists( self::LIGHTWEIGHT_API_CONFIG_FILE_PATH ) ) {
+			return;
+		}
+		?>
+			<div class="notice notice-error">
```

Comment:
Slight rewrite:

```
<div class="notice notice-error">
				<p>
					<?php _e( 'Newspack Campaigns requires a custom configuration file, which is missing. Please create this file following instructions found ', 'newspack-popups' ); ?>
					<a href="https://github.com/Automattic/newspack-popups/blob/master/api/README.md">
						<?php _e( 'here.', 'newspack-popups' ); ?>
					</a>
				</p>
			</div>
```

Reply:
Done in d4d9a2b82c5f97ce009aa3b5fb5719d5e58b34c0